src/reader.rs

```diff
@@ -144,6 +144,160 @@ impl Reader {
         Ok(None)
     }
 
+    pub(crate) async fn multi_get_with_options<K: AsRef<[u8]>>(
```

Comment:
Given that this looks mostly like a superset of the functionality provided in get_with_options, WDYT about having get_with_options just call multi_get_with_options with a single key? I don't think it would add significant overhead, and might alleviate some of @flaneur2020's concern about implementation complexity.